### ISSUE TITLE
CS8672 should not warn when returning in local function

### ISSUE THREAD

**Issue by JohanLarsson** [dotnet/roslyn#44289]
```cs
class C
{
    bool M([NotNullWhen(true)] out string? p)
    {
        p = null;
        return false;

        bool Inner()
        {
            // ↓ CS8762	Parameter 'p' may not have a null value when exiting with 'true'
            return true;
        }
    }
}
```

**Comment by RikkiGibson** [comment]
I can't reproduce this in [SharpLab](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxASwDYB8DEA7AVwwwmAzgAI49TyBYAKAAEAGCpgRgDoARNCAOZ4A9rDQBjKFwDCwgCZwAgjQwBPKGigBuRoyYAmCtMYBvRhQsVgw4RgoBZABQBtAHLCYrohgDqAC2pHGAQCOABKAF0KYQIYdg4WAH4KAAcw80szBksc1IoAXgpCYh1s3IsmAHYKADMIDCg4Uozy61sKAEk8PERHdLLyrPLygHoRikBkwiMAZQAOSoA2fQBIAAUIBAgwOHgECgByFP2KSFUijwo/CAA3Sggi7wpr+tCKAHcAvCoADzQYNDwAnefz8B2CoX2LWGFWq4KaUJyAF8ociGIigA=).
